### PR TITLE
refactor(db): Add TenantId field to the KafkaStore struct

### PR DESCRIPTION
**Pull request by subhajit20**
## Type of Change

Add TenantId field to the KafkaStore struct
#4496 

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Add TenantId field to the KafkaStore struct
- Add the tenant_id parameter to the constructor of Kafkastore 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

- no testable changes except CI checks

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/db/kafka_store.rs [hunk 1]
```diff
@@ -72,17 +72,22 @@ use crate::{
     },
 };
 
+#[derive(Clone)]
+pub struct TenantID(String)
```
Comment (lsampras) [inline crates/router/src/db/kafka_store.rs:76]: Can we add Serialize/Deserialize macros for this?

File: crates/router/src/db/kafka_store.rs [hunk 2]
```diff
@@ -72,17 +72,22 @@ use crate::{
...
 #[derive(Clone)]
 pub struct KafkaStore {
     kafka_producer: KafkaProducer,
     pub diesel_store: Store,
+    pub tenant_id: TenantID
 }
 
 impl KafkaStore {
-    pub async fn new(store: Store, kafka_producer: KafkaProducer) -> Self {
+    pub async fn new(store: Store, kafka_producer: KafkaProducer, tenant_id: TenantID) -> Self {
```
Comment (lsampras) [inline crates/router/src/db/kafka_store.rs:86]: @subhajit20 we'll need to update the usages for this function as well...

**Review by lsampras — COMMENTED** [review]
The approach looks good, have left some comments..

**Comment by lsampras** [comment]
@subhajit20 Thanks for incorporating my comments,
There's some linting issues tho, will merge this once that is fixed...

**Comment by subhajit20** [comment]
> @subhajit20 Thanks for incorporating my comments, There's some linting issues tho, will merge this once that is fixed...

Hey Done that.

**Review by lsampras — APPROVED** [review]
@subhajit20 Thanks for the contribution,

These changes look good to me.

**Pull request merged**